core/src/test/java/org/elasticsearch/index/mapper/NestedObjectMapperTests.java

```diff
@@ -525,35 +525,42 @@ public void testParentObjectMapperAreNested() throws Exception {
     }
 
     public void testLimitNestedDocsDefaultSettings() throws Exception{
-        MapperService mapperService = createIndex("test1", Settings.builder().build()).mapperService();
+        Settings settings = Settings.builder().build();
+        MapperService mapperService = createIndex("test1", settings).mapperService();
         String mapping = XContentFactory.jsonBuilder().startObject().startObject("type").startObject("properties")
             .startObject("nested1").field("type", "nested").endObject()
             .endObject().endObject().endObject().string();
         DocumentMapper docMapper = mapperService.documentMapperParser().parse("type", new CompressedXContent(mapping));
+        long defaultMaxNoNestedDocs = MapperService.INDEX_MAPPING_NESTED_DOCS_LIMIT_SETTING.get(settings);
 
-        // parsing a doc with 3 nested objects succeeds
+        // parsing a doc with No. nested objects > defaultMaxNoNestedDocs fails
         XContentBuilder docBuilder = XContentFactory.jsonBuilder();
         docBuilder.startObject();
         {
             docBuilder.startArray("nested1");
             {
-                docBuilder.startObject().field("field1", "11").field("field2", "21").endObject();
-                docBuilder.startObject().field("field1", "12").field("field2", "22").endObject();
-                docBuilder.startObject().field("field1", "13").field("field2", "23").endObject();
+                for(int i=0; i<=defaultMaxNoNestedDocs; i++) {
+                    docBuilder.startObject().field("f", i).endObject();
+                }
             }
             docBuilder.endArray();
         }
         docBuilder.endObject();
         SourceToParse source1 = SourceToParse.source("test1", "type", "1", docBuilder.bytes(), XContentType.JSON);
-        ParsedDocument doc = docMapper.parse(source1);
-        assertThat(doc.docs().size(), equalTo(4));
+        MapperParsingException e = expectThrows(MapperParsingException.class, () -> docMapper.parse(source1));
+        assertEquals(
+            "The number of nested documents has exceeded the allowed limit of [" + defaultMaxNoNestedDocs
+                + "]. This limit can be set by changing the [" + MapperService.INDEX_MAPPING_NESTED_DOCS_LIMIT_SETTING.getKey()
+                + "] index level setting.",
+            e.getMessage()
+        );
     }
 
     public void testLimitNestedDocs() throws Exception{
         // setting limit to allow only two nested objects in the whole doc
-        long nested_docs_limit = 2L;
+        long maxNoNestedDocs = 2L;
```

Comment:
just as feedback, nothing to change really, but I liked the previous variable name better ;-)

Reply:
@cbuescher thanks for the feedback, I thought in the previous variable names I did not follow Java camel case standards for variable names (was more of python style)

Reply:
I see, and you are right, camel case is preferred. I probably misread the "NoNestedDocs" part of the name as "no nested docs" and that confused me for a second, but either way is fine.

Reply:
@cbuescher thanks Christoph, you are right`NoNestedDocs` is confusing, I will avoid using `No` for "number of" in the future.